### PR TITLE
optimize: Local variable 'map' is redundant and check queue offer return value

### LINKED ISSUE #3187: optimize: Local variable 'map' is redundant  and check queue offer return value
https://github.com/seata/seata/blob/6c0c75939dedbf19c4a21a8ac7aa0fb98879a8fd/core/src/main/java/io/seata/core/rpc/netty/AbstractNettyRemotingClient.java#L156

  // put message into basketMap
            ConcurrentHashMap<String, BlockingQueue<RpcMessage>> map = basketMap;  -> map is redundant
            BlockingQueue<RpcMessage> basket = map.get(serverAddress);
            if (basket == null) {
                map.putIfAbsent(serverAddress, new LinkedBlockingQueue<>());
                basket = map.get(serverAddress);
            }
            basket.offer(rpcMessage); -> check return value

### PR DESCRIPTION
**Pull request by everyhook1**
### Ⅰ. Describe what this PR did
https://github.com/seata/seata/blob/6c0c75939dedbf19c4a21a8ac7aa0fb98879a8fd/core/src/main/java/io/seata/core/rpc/netty/AbstractNettyRemotingClient.java#L156

  // put message into basketMap
            ConcurrentHashMap<String, BlockingQueue<RpcMessage>> map = basketMap;
            BlockingQueue<RpcMessage> basket = map.get(serverAddress);
            if (basket == null) {
                map.putIfAbsent(serverAddress, new LinkedBlockingQueue<>());
                basket = map.get(serverAddress);
            }
            basket.offer(rpcMessage);

1.remove redundant local variable 'map';
2.check the BlockingQueue offer return value;
### Ⅱ. Does this pull request fix one issue?

fixes #3187

### CODE REVIEW

File: core/src/main/java/io/seata/core/rpc/netty/AbstractNettyRemotingClient.java [hunk 1]
```diff
@@ -152,13 +152,14 @@ public Object sendSyncRequest(Object msg) throws TimeoutException {
             futures.put(rpcMessage.getId(), messageFuture);
 
             // put message into basketMap
-            ConcurrentHashMap<String, BlockingQueue<RpcMessage>> map = basketMap;
-            BlockingQueue<RpcMessage> basket = map.get(serverAddress);
-            if (basket == null) {
-                map.putIfAbsent(serverAddress, new LinkedBlockingQueue<>());
-                basket = map.get(serverAddress);
+            basketMap.computeIfAbsent(serverAddress, k -> new LinkedBlockingQueue<>());
+            if (!basketMap.get(serverAddress).offer(rpcMessage)) {
```
Comment (wangliang181230) [inline core/src/main/java/io/seata/core/rpc/netty/AbstractNettyRemotingClient.java:156]:
```
BlockingQueue<RpcMessage> basket = basketMap.computeIfAbsent(serverAddress, k -> new LinkedBlockingQueue<>());
if (!basket .offer(rpcMessage)) {
```

File: core/src/main/java/io/seata/core/rpc/netty/AbstractNettyRemotingClient.java [hunk 2]
```diff
@@ -152,13 +152,14 @@ public Object sendSyncRequest(Object msg) throws TimeoutException {
...
+                if(LOGGER.isErrorEnabled()) {
```
Comment (wangliang181230) [inline core/src/main/java/io/seata/core/rpc/netty/AbstractNettyRemotingClient.java:157]: not need check if is error log.

File: core/src/main/java/io/seata/core/rpc/netty/AbstractNettyRemotingClient.java [hunk 3]
```diff
@@ -152,13 +152,10 @@ public Object sendSyncRequest(Object msg) throws TimeoutException {
             futures.put(rpcMessage.getId(), messageFuture);
 
             // put message into basketMap
-            ConcurrentHashMap<String, BlockingQueue<RpcMessage>> map = basketMap;
-            BlockingQueue<RpcMessage> basket = map.get(serverAddress);
-            if (basket == null) {
-                map.putIfAbsent(serverAddress, new LinkedBlockingQueue<>());
-                basket = map.get(serverAddress);
+            BlockingQueue<RpcMessage> basket = basketMap.computeIfAbsent(serverAddress, k -> new LinkedBlockingQueue<>());
+            if (basket.offer(rpcMessage)) {
```
Comment (wangliang181230) [inline core/src/main/java/io/seata/core/rpc/netty/AbstractNettyRemotingClient.java:156]: lost the `!`

File: core/src/main/java/io/seata/core/rpc/netty/AbstractNettyRemotingClient.java (hunk 2 above)
Comment (wangliang181230) [reply core/src/main/java/io/seata/core/rpc/netty/AbstractNettyRemotingClient.java]: I mean, `if` is not required, but log is still required.

**Comment by a364176773** [comment]
请绑定一下githubid到你本地的git插件上

File: core/src/main/java/io/seata/core/rpc/netty/AbstractNettyRemotingClient.java [hunk 4]
```diff
@@ -152,13 +152,12 @@ public Object sendSyncRequest(Object msg) throws TimeoutException {
...
 
             // put message into basketMap
-            ConcurrentHashMap<String, BlockingQueue<RpcMessage>> map = basketMap;
-            BlockingQueue<RpcMessage> basket = map.get(serverAddress);
-            if (basket == null) {
-                map.putIfAbsent(serverAddress, new LinkedBlockingQueue<>());
-                basket = map.get(serverAddress);
+            basketMap.putIfAbsent(serverAddress, new LinkedBlockingQueue<>());
+            if (!basketMap.get(serverAddress).offer(rpcMessage)) {
+                LOGGER.error("put message into basketMap offer failed, serverAddress:{},rpcMessage:{}",
```
[the remark this answers is not in the document]
Comment (wangliang181230) [reply core/src/main/java/io/seata/core/rpc/netty/AbstractNettyRemotingClient.java:157]: 可以不用if判断的。error日志基本上需要记录，另一个，日志也使用了占位符，没什么问题。

**Comment by everyhook1** [comment]
> 请绑定一下githubid到你本地的git插件上

**Review by a364176773 — APPROVED** [review]
LGTM

**Review by slievrly — APPROVED** [review]
LGTM

**Comment by a364176773** [comment]
resolve code conflicts

**Pull request merged**